linera-graphql-client/src/service.rs

```diff
@@ -0,0 +1,185 @@
+    pub struct Notification {
+        pub chain_id: ChainId,
+        pub reason: Reason,
+    }
+
+    #[derive(Serialize, Deserialize, Debug, Clone, PartialEq)]
+    #[allow(clippy::enum_variant_names)]
+    pub enum Reason {
+        NewBlock {
+            height: BlockHeight,
+            hash: CryptoHash,
+        },
+        NewIncomingMessage {
+            origin: Origin,
+            height: BlockHeight,
+        },
+        NewRound {
+            height: BlockHeight,
+            round: RoundNumber,
+        },
+    }
+}
+
+#[cfg(not(target_arch = "wasm32"))]
+mod types {
+    pub use linera_chain::data_types::{
+        notifications::{Notification, Reason},
+        Event, Origin,
+    };
+    pub use linera_execution::{
+        committee::Epoch, ApplicationId, Message, Operation, UserApplicationDescription,
+    };
+}
+
+pub use types::*;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone"
+)]
+pub struct Chains;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone, PartialEq"
+)]
+pub struct Applications;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone, PartialEq"
+)]
+pub struct Blocks;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone, PartialEq"
+)]
+pub struct Block;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone, PartialEq"
+)]
+pub struct Notifications;
+
+#[derive(GraphQLQuery)]
+#[graphql(
+    schema_path = "graphql/service_schema.graphql",
+    query_path = "graphql/service_requests.graphql",
+    response_derives = "Debug, Serialize, Clone"
+)]
+pub struct Transfer;
+
+#[cfg(not(target_arch = "wasm32"))]
+mod from {
+    use super::*;
+    use linera_chain::data_types::{ExecutedBlock, HashedValue, IncomingMessage, OutgoingMessage};
+
+    impl From<block::BlockBlockValueExecutedBlockBlockIncomingMessages> for IncomingMessage {
+        fn from(val: block::BlockBlockValueExecutedBlockBlockIncomingMessages) -> Self {
+            IncomingMessage {
+                origin: val.origin,
+                event: val.event,
+            }
+        }
+    }
+
+    impl From<block::BlockBlockValueExecutedBlockBlock> for linera_chain::data_types::Block {
+        fn from(val: block::BlockBlockValueExecutedBlockBlock) -> Self {
+            let incoming_messages = val
```

Comment:
Would it make sense to destructure `val` here, so that the compiler would tell us if we missed any fields? Or is `BlockBlockValueExecutedBlockBlock` auto-generated from `Block` anyway?

Reply:
it's generated from the GraphQL query so it can miss some fields. It would be a good idea to destruct it.